core/src/main/java/io/questdb/griffin/WhereClauseParser.java

```diff
@@ -125,11 +127,36 @@ private boolean analyzeEquals0(AliasTranslator translator, IntrinsicModel model,
                                         model.keyValues.add(value);
                                         model.keyValuePositions.add(b.position);
                                     }
+                                    node.intrinsicValue = IntrinsicModel.TRUE;
                                 } else {
+                                    if (model.keyExcludedValues.contains(value)) {
+                                        if (model.keyExcludedValues.size() > 1) {
+                                            model.keyExcludedValues.remove(value);
+                                            int removedIndex = model.keyExcludedValues.remove(value);
+                                            if (removedIndex > -1) {
+                                                model.keyExcludedValuePositions.removeIndex(index);
+                                            }
+                                        } else {
+                                            model.keyExcludedValues.clear();
+                                            model.keyExcludedValuePositions.clear();
+                                        }
+                                        tempNodes.clear();
+                                        for (int i = 0; i < keyExclNodes.size(); i++) {
```

Comment:
the loops should not be calling `size()` on every iteration. Somewhere else you did:

```sql
for (int n = 0, k = keyExclNodes.size(); n < k; n++) {
```

which is much better